src/seq/fastq/quality.jl

```diff
@@ -6,221 +6,69 @@
 # This file is a part of BioJulia.
 # License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
 
-"A `QualityEncoding` value holds a set of encoding compatible with a string"
-bitstype 16 QualityEncoding
+immutable QualityEncoding
+    # valid ASCII character range
+    ascii::NTuple{2,UInt8}
 
-Base.convert(::Type{QualityEncoding}, nt::UInt16) = reinterpret(QualityEncoding, nt)
-Base.convert(::Type{UInt16}, nt::QualityEncoding) = reinterpret(UInt16, nt)
+    # valid base quality range
+    qual::NTuple{2,Int8}
 
-function Base.:|(a::QualityEncoding, b::QualityEncoding)
-    return convert(QualityEncoding, convert(UInt16, a) | convert(UInt16, b))
+    function QualityEncoding(ascii::Range{Char}, qual::Range{Int})
+        if length(ascii) != length(qual)
+            throw(ArgumentError("the ranges of ASCII and base quality don't match"))
+        end
+        return new((first(ascii), last(ascii)), (first(qual), last(qual)))
+    end
 end
 
-function Base.:&(a::QualityEncoding, b::QualityEncoding)
-    return convert(QualityEncoding, convert(UInt16, a) & convert(UInt16, b))
+# Return the offset of converting a base quality to an ASCII code (quality + offset).
+function ascii_offset(encoding::QualityEncoding)
```

Comment:
I think this function will not be a bottleneck because it can be called outside of a loop.